From ca6a98f8ef9d714aebef4d2637d72ce78ddaaea1 Mon Sep 17 00:00:00 2001
From: Alex Richardson <Alexander.Richardson@cl.cam.ac.uk>
Date: Mon, 23 Nov 2020 12:58:00 +0000
Subject: [PATCH] Use git-filter-branch --index-filter instead of --tree-filter

Using an index filter is much faster since git does not have to perform
a full checkout for every commit in the range that needs to be rewritten.

Fixes #435
---
 lib/git-subrepo | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/git-subrepo b/lib/git-subrepo
index 6a066a5..4ab1edb 100755
--- a/lib/git-subrepo
+++ b/lib/git-subrepo
@@ -816,8 +816,8 @@ subrepo:branch() {
   o "Remove the .gitrepo file from $first_gitrepo_commit..$branch"
   local filter=$branch
   [[ $first_gitrepo_commit ]] && filter=$first_gitrepo_commit..$branch
-  FAIL=false RUN git filter-branch -f --prune-empty --tree-filter \
-    "rm -f .gitrepo" "$filter"
+  FAIL=false RUN git filter-branch -f --prune-empty --index-filter \
+    'git rm --cached --ignore-unmatch .gitrepo' "$filter"
 
   git:create-worktree "$branch"
 
